### PR TITLE
feat: add a search field to server bans

### PR DESCRIPTION
**Pull request by janderedev**
## Please make sure to check the following tasks before opening and submitting a PR

* [x] I understand and have followed the [contribution guide](https://github.com/revoltchat/revolt/discussions/282)
* [x] I have tested my changes locally and they are working as intended
* [x] These changes do not have any notable side effects on other Revolt projects
* [ ] (optional) I have opened a pull request on [the translation repository](https://github.com/revoltchat/translations)
* [x] I have included screenshots to demonstrate my changes

Adds a search bar to the server ban list:
image

(I copypasted the search bar from the member list, hence the lack of localization)

### CODE REVIEW

**Comment by EnokiUN** [comment]
True and holy in nature.

EGN 12/10

**Review by EnokiUN — DISMISSED** [review]

**Comment by insertish** [comment]
Conflicting

File: src/pages/settings/server/Bans.tsx [hunk 1]
```diff
@@ -75,24 +83,33 @@ export const Bans = observer(({ server }: Props) => {
...
-                            />
-                        )}
+                        itemContent={(index) =>
+                            (!query ||
+                                data.users
+                                    .find(
+                                        (u) =>
+                                            u._id == data.bans[index]._id.user,
+                                    )
+                                    ?.username.toLowerCase()
```
Comment (insertish) [inline src/pages/settings/server/Bans.tsx:93]: This is a pretty big performance penalty which should be memo-ized.

Comment (janderedev) [reply src/pages/settings/server/Bans.tsx]: Like this?

Comment (insertish) [reply src/pages/settings/server/Bans.tsx]: `useMemo`, `useCallback` just prevents the function from being recreated, while `useMemo` keeps the result between renders and also you don't call it every time you render an item

Comment (janderedev) [reply src/pages/settings/server/Bans.tsx]: TIL

**Pull request closed without merging**